crates/bevy_transform/src/components/transform.rs

```diff
@@ -366,6 +399,67 @@ impl Transform {
         self.rotation = Quat::from_mat3(&Mat3::from_cols(right, up, back));
     }
 
+    /// Rotates this [`Transform`] so that the `main_axis` vector, reinterpreted in local coordinates, points
+    /// in the given `main_direction`, while `secondary_axis` points towards `secondary_direction`.
+    ///
+    /// For example, if a spaceship model has its nose pointing in the X-direction in its own local coordinates
+    /// and its dorsal fin pointing in the Y-direction, then `align(Vec3::X, v, Vec3::Y, w)` will make the spaceship's
+    /// nose point in the direction of `v`, while the dorsal fin does its best to point in the direction `w`.
+    ///
+    /// More precisely, the [`Transform::rotation`] produced will be such that:
+    /// * applying it to `main_axis` results in `main_direction`
+    /// * applying it to `secondary_axis` produces a vector that lies in the half-plane generated by `main_direction` and
+    /// `secondary_direction` (with positive contribution by `secondary_direction`)
+    ///
+    /// [`Transform::look_to`] is recovered, for instance, when `main_axis` is `Vec3::NEG_Z` (the [`Transform::forward`]
+    /// direction in the default orientation) and `secondary_axis` is `Vec3::Y` (the [`Transform::up`] direction in the default
+    /// orientation). (Failure cases may differ somewhat.)
+    ///
+    /// In some cases a rotation cannot be constructed. Another axis will be picked in those cases:
+    /// * if `main_axis` or `direction` is zero, `Vec3::X` takes its place
+    /// * if `secondary_axis` or `secondary_direction` is zero, `Vec3::Y` takes its place
+    /// * if `main_axis` is parallel with `secondary_axis` or `main_direction` is parallel with `secondary_direction`,
+    /// a rotation is constructed which takes `main_axis` to `main_direction` but ignores the secondary counterparts
+    /// (i.e. is otherwise unspecified)
+    #[inline]
+    pub fn align(
+        &mut self,
+        main_axis: Vec3,
+        main_direction: Vec3,
+        secondary_axis: Vec3,
+        secondary_direction: Vec3,
+    ) {
+        let main_axis = main_axis.try_normalize().unwrap_or(Vec3::X);
+        let main_direction = main_direction.try_normalize().unwrap_or(Vec3::X);
+        let secondary_axis = secondary_axis.try_normalize().unwrap_or(Vec3::Y);
+        let secondary_direction = secondary_direction.try_normalize().unwrap_or(Vec3::Y);
+
+        // The solution quaternion will be constructed in two steps.
+        // First, we start with a rotation that takes `handle` to `direction`.
+        let first_rotation = Quat::from_rotation_arc(main_axis, main_direction);
+
+        // Let's follow by rotating about the `direction` axis so that the image of `weak_handle`
+        // is taken to something that lies in the plane of `direction` and `weak_direction`. Since
+        // `direction` is fixed by this rotation, the first criterion is still satisfied.
```

Comment:
I really appreciate your approach to documenting this feature! Nit pick, but I think you may have changed the variable names at some point, `handle` and `weak_handle` no longer exist.

Reply:
I really thought I caught all these...